src/PJV.ts

```diff
@@ -0,0 +1,528 @@
+				errors.push(`URL not valid for ${name}: ${obj.web}`);
+			}
+		}
+	} else {
+		errors.push(`Type for field ${name} should be a string or an object`);
+	}
+	return errors;
+};
+
+/**
+ * Validate 'people' fields, which can be an object like this:
+ *
+ * {
+ *   "name" : "Barney Rubble"
+ *   "email" : "b@rubble.com",
+ *   "url" : "http://barnyrubble.tumblr.com/"
+ *   "web" : "http://barnyrubble.tumblr.com/"
+ * }
+ *
+ * Or a single string like this:
+ * "Barney Rubble <b@rubble.com> (http://barnyrubble.tumblr.com/)
+ * Or an array of either of the above.
+ */
+const validatePeople = (name: string, obj: People): string[] => {
+	const errors: string[] = [];
+
+	function validatePerson(obj: string | Person) {
+		if (typeof obj == "string") {
+			const authorRegex = /^([^<\(\s]+[^<\(]*)?(\s*<(.*?)>)?(\s*\((.*?)\))?/;
+			const authorFields = authorRegex.exec(obj);
+			if (authorFields) {
+				const authorName = authorFields[1];
+				const authorEmail = authorFields[3];
+				const authorUrl = authorFields[5];
+				validatePerson({
+					name: authorName,
+					email: authorEmail,
+					url: authorUrl,
+				});
+			} else {
+				errors.push(`Unable to parse person string: ${obj}`);
+			}
+		} else if (typeof obj == "object") {
+			if (!obj.name) {
+				errors.push(`${name} field should have name`);
+			}
+			if (obj.email && !emailFormat.test(obj.email)) {
+				errors.push(`Email not valid for ${name}: ${obj.email}`);
+			}
+			if (obj.url && !urlFormat.test(obj.url)) {
+				errors.push(`URL not valid for ${name}: ${obj.url}`);
+			}
+			if (obj.web && !urlFormat.test(obj.web)) {
+				errors.push(`URL not valid for ${name}: ${obj.web}`);
+			}
+		} else {
+			errors.push("People field must be an object or a string");
+		}
+	}
+
+	if (obj instanceof Array) {
+		for (var i = 0; i < obj.length; i++) {
+			validatePerson(obj[i]);
+		}
+	} else {
+		validatePerson(obj);
+	}
+	return errors;
+};
+
+/**
+ * Format for license(s) and repository(s):
+ * url as a string
+ * or
+ * object with "type" and "url"
+ * or
+ * array of objects with "type" and "url"
+ */
+const validateUrlTypes = (
+	name: string,
+	obj: string | UrlType | UrlType[],
+): string[] => {
+	const errors: string[] = [];
+
+	const validateUrlType = (obj: UrlType) => {
+		if (!obj.type) {
+			errors.push(`${name} field should have type`);
+		}
+		if (!obj.url) {
+			errors.push(`${name} field should have url`);
+		}
+	};
+
+	if (typeof obj === "string") {
+		if (!urlFormat.test(obj)) {
+			errors.push(`URL not valid for ${name}: ${obj}`);
+		}
+	} else if (obj instanceof Array) {
+		for (var i = 0; i < obj.length; i++) {
+			validateUrlType(obj[i]);
+		}
+	} else if (typeof obj === "object") {
+		validateUrlType(obj);
+	} else {
+		errors.push(`Type for field ${name} should be a string or an object`);
+	}
+
+	return errors;
+};
+
+// TODO: Should we consider deprecating the PJV named export in favor of directly
+// exporting the functions and constants?
```

Comment:
> I did put a TODO in though to consider deprecating the one named PJV export, in favor of directly exporting the functions and constants. This would make the API more tree-shakable and (in my opinion) more straightforward. But that should be a separate change, if implemented.

Agreed, I think that'd be a good followup.

Reply:
Opened https://github.com/JoshuaKGoldberg/package-json-validator/issues/124 to track